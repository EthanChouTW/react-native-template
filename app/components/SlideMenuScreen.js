import React, { PureComponent } from 'react';
import { DrawerItems } from 'react-navigation';
import { View, Text, StyleSheet } from 'react-native';
import Theme from '../utils/styleCollection';
import { version } from '../../package.json';

const styles = StyleSheet.create({
  drawer: {
    flex: 1,
    backgroundColor: 'black'
  },
  userInfoContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    padding: 16,
    height: 60
  },
  avatar: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  userName: {
    color: 'white',
    fontSize: 16
  },
  email: {
    color: Theme.hbGray,
    fontSize: 14
  },
  versioning: {
    color: Theme.hbGray,
    padding: 16,
    fontSize: 16
  }
});

export default class SlideMenuScreen extends PureComponent {
  render() {
    return (
      <View style={styles.drawer}>
        <View style={{ flex: 1 }}>
          <DrawerItems {...this.props} />
        </View>
        <Text style={styles.versioning}>
          version: v{version}
        </Text>
      </View>
    );
  }
}
