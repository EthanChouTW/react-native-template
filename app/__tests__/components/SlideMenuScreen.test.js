import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import SlideMenuScreen from '../../components/SlideMenuScreen';

describe('UserProfileMenuComponent', () => {
  const routes = [
    {
      key: 'UserProfile',
      routeName: 'UserProfile'
    },
    {
      key: 'Home',
      routeName: 'Home'
    },
    {
      key: 'InventoryManagementScreen',
      routeName: 'InventoryManagementScreen'
    },
    {
      key: 'JobHistory',
      routeName: 'JobHistory'
    },
    {
      key: 'Logout',
      routeName: 'Logout'
    }
  ];
  const props = {
    renderIcon: jest.fn(),
    getLabel: jest.fn(),
    navigation: {
      state: {
        routes
      }
    },
    items: routes
  };
  test('renders correctly', () => {
    const snapshot = renderer.create(<SlideMenuScreen {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
