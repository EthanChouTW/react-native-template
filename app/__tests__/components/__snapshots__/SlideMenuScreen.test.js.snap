// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileMenuComponent renders correctly 1`] = `
<
  style={
    Object {
      "backgroundColor": "black",
      "flex": 1,
    }
  }
>
  <
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <
      style={
        Array [
          Object {
            "marginTop": 20,
            "paddingVertical": 4,
          },
          undefined,
        ]
      }
    >
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </>
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </>
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </>
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </>
      <
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </>
    </>
  </>
  <
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#E5E5E5",
        "fontSize": 16,
        "padding": 16,
      }
    }
  >
    version: v
    0.9.0
  </>
</>
`;
